### PR TITLE
Add autoreconnect feature

### PR DESCRIPTION
**Pull request by Shir0kamii**
Hi,

I've implemented a new option to the `GraphiteClient` object. It will try to reconnect automatically to the server when `autoreconnect` is set to `True`. Of course the default is `False`.

By default, it will try to reconnect 3 times with 1 second of interval. If you have opinion on these default, let me know as I'm not sure if these are the best. After these tries, an error will be thrown like before if `GraphiteClient` didn't reconnect successfully.

I splited the `_send` function to allow more modularity, I hope you don't mind.

Finally, I added tests for this function, based on what you've done for async feature.

I hope you like the idea, but feel free to tell me what dislike you.

### CODE REVIEW

**Comment by Shir0kamii** [comment]
@daniellawrence Any chance that you look at this pull request soon ?

**Comment by daniellawrence** [comment]
I'll review this tonight, thanks for the help so far :)

File: graphitesend/graphitesend.py [hunk 1]
```diff
@@ -221,35 +243,55 @@ def _send(self, message):
                 "Socket was not created before send"
             )
 
+        fct = self._send
```
Comment (daniellawrence) [inline graphitesend/graphitesend.py]: something more descriptive than "tct" would be nice.

**Comment by daniellawrence** [comment]
* Added a comment about renaming "fct" to something more descriptive.
* Also flake8 triggered too many blank lines - `./graphitesend/graphitesend.py:435:1: E303 too many blank lines (3)`

Both should be easy fixes :)

**Comment by Shir0kamii** [comment]
@daniellawrence I changed what you asked

**Comment by Shir0kamii** [comment]
@daniellawrence Are you still working on graphitesend lately ?

**Comment by Shir0kamii** [comment]
@daniellawrence Thank you for merging ! Would it be possible that you publish the changes on Pypi ?

**Comment by daniellawrence** [comment]
Hey @Shir0kamii, 
  Released your changes (thanks again!) as 0.6.0, Also added in https://github.com/daniellawrence/graphitesend/blob/master/HOW_TO_RELEASE.md, which should help speed up this in the future.
#42 has the details.

Thanks again!

**Pull request merged**